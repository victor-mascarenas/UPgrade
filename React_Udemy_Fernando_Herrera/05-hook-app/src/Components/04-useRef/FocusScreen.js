import React, { useRef } from 'react';
import '../02-useEffect/effects.css';

const FocusScreen = () => {
    const inputRef = useRef();

    const btnOnClick = () => {
        // document.querySelector('input').select();
        inputRef.current.select();
    }

    return (
        <div>
            <h1>Focus screen</h1>
            <hr/>
            <input ref={inputRef} className='form-control' placeholder='Su nombre'/>
            <button className='btn btn-outline-primary mt-3' onClick={btnOnClick}>Focus</button>
        </div>
    );
};

export default FocusScreen;