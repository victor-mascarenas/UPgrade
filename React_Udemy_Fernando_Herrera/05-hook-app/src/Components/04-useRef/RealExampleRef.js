import React, { useState } from 'react';
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';
import '../02-useEffect/effects.css';

export const RealExampleRef = () => {
    const [show, setShow] = useState(false);
    
    const buttonOnClick = () => {
        setShow(!show);
    };
    
    return (
        <div>
            <h1>Real example</h1>
            <hr/>
            {show && <MultipleCustomHooks/>}
            <button className='btn btn-primary mt-3' onClick={buttonOnClick}>Show/Hide</button>
        </div>
    );
};
