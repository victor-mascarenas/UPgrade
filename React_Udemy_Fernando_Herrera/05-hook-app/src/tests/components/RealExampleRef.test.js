import React from "react";
import {shallow} from 'enzyme';
import {renderHook, act} from '@testing-library/react-hooks';
import { RealExampleRef } from "../../Components/04-useRef/RealExampleRef";

describe('Pruebas en RealExampleRef', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<RealExampleRef/>);
    });

    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar el componente MultipleCustomHooks', () => {
        const btn = wrapper.find('button');
        btn.simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBeTruthy();
    });
    test('No debe de mostrar el componente MultipleCustomHooks', () => {
        expect(wrapper.find('MultipleCustomHooks').exists()).toBeFalsy();
    });
});
