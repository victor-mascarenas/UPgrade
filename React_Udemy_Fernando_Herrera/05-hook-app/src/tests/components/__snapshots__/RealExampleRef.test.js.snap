// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en RealExampleRef Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Real example
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
